src/m365/entra/commands/group/group-set.spec.ts

```diff
@@ -0,0 +1,353 @@
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerIds: ownerIds.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if ownerUserNames contains invalid user principal name', async () => {
+    const ownerUserNames = ['john.doe@contoso.com', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerUserNames: ownerUserNames.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if memberIds contains invalid GUID', async () => {
+    const memberIds = ['7167b488-1ffb-43f1-9547-35969469bada', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', memberIds: memberIds.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if memberUserNames contains invalid user principal name', async () => {
+    const memberUserNames = ['john.doe@contoso.com', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', memberUserNames: memberUserNames.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if visibility contains invalid value', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', visibility: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if id is a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: groupId, newDisplayName: 'Cli group' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if id is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: 'foo', newDisplayName: 'Cli group' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with ids', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerIds: userIds.join(','), memberIds: userIds.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with user names', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerUserNames: userUpns.join(','), memberUserNames: userUpns.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('successfully updates group specified by id', async () => {
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}` && 
+        JSON.stringify(opts.data) === JSON.stringify({
+          "displayName": '365 group',
+          "description": 'Microsoft 365 group',
+          "mailNickName": 'Microsoft365Group',          
+          "visibility": 'Public'
+        })) {
```

Comment:
For this and all following tests, this is not really a good way to check the request body. If the request body mismatches, we'll get an 'invalid request' error and it's very difficult to tell whats wrong.
A better way is that we check the request body in an assert statement. This will give the expected and the current output in an error message. Displaying the difference between the two. It will make debugging the test a lot easier.
You can find an example here: https://github.com/pnp/cli-microsoft365/blob/d20a60e90737fbff0c78105b88dab82611b0a442/src/m365/entra/commands/user/user-guest-add.spec.ts#L147-L159